### PR TITLE
avoid overwrite old database value when republish smart contract

### PR DESCRIPTION
**Pull request by learnforpractice**

### CODE REVIEW

File: contracts/currency/currency.cpp [hunk 1]
```diff
@@ -39,7 +39,12 @@ using namespace TOKEN_NAME;
 
 extern "C" {
     void init()  {
-       store_account( N(currency), account( currency_tokens(1000ll*1000ll*1000ll) ) );
+       account owned_account;
+       if ( accounts::get( owned_account, N(currency) )) {
+          //currency account already exist, do nothing
+       } else {
```
Comment (spicynoodle) [inline contracts/currency/currency.cpp]: why not `if( !account::get(owned_account, N(currency) ))` no need else

**Comment by learnforpractice** [comment]
I don’t like “not". :)

**Comment by learnforpractice** [comment]
Ok, I can explain it more clearly: It's my own style to improve  code readability.I’m heavily influence by python.On 16 Nov 2017, at 2:55 PM, Retriever <notifications@github.com> wrote:@spicynoodle commented on this pull request.

In contracts/currency/currency.cpp:
> @@ -39,7 +39,12 @@ using namespace TOKEN_NAME;
 
 extern "C" {
     void init()  {
-       store_account( N(currency), account( currency_tokens(1000ll*1000ll*1000ll) ) );
+       account owned_account;
+       if ( accounts::get( owned_account, N(currency) )) {
+          //currency account already exist, do nothing
+       } else {
why not if( !account::get(owned_account, N(currency) )) no need else—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by learnforpractice** [comment]
Okay, I changed it as below:
```
       account owned_account;
       //Initialize currency account only if it does not exist
       if ( !accounts::get( owned_account, N(currency) )) {
          store_account( N(currency), account( currency_tokens(1000ll*1000ll*1000ll) ) );
       }
```
Also in storage.cpp and infinite.cpp

**Pull request merged**